templates/repo/view_file.tmpl

```diff
@@ -58,6 +58,13 @@
 						<a class="ui mini basic button unescape-button" style="display: none;">{{.locale.Tr "repo.unescape_control_characters"}}</a>
 						<a class="ui mini basic button escape-button">{{.locale.Tr "repo.escape_control_characters"}}</a>
 					{{end}}
+					<button class="ui mini basic button compact tooltip" id="copy-file-content" data-content="{{.locale.Tr "copy_content"}}" aria-label="{{.locale.Tr "copy_content"}}"
+						{{if or (.IsMarkup) (.IsRenderedHTML) (not .IsTextSource)}}
+							disabled
+						{{end}}
+					>
```

Comment:
```suggestion
					<button class="ui mini basic button compact tooltip" id="copy-file-content" data-content="{{.locale.Tr "copy_content"}}" aria-label="{{.locale.Tr "copy_content"}}"{{if or (.IsMarkup) (.IsRenderedHTML) (not .IsTextSource)}} disabled{{end}}>
```

This extra whitespace would be visible in HTML otherwise.

Reply:
Also, I think parens around `(.IsMarkup)` and `(.IsRenderedHTML)` can be removed, but please test it.